packages/hoppscotch-common/src/services/spotlight/searchers/environment.searcher.ts

```diff
@@ -318,6 +363,31 @@ export class SwitchEnvSpotlightSearcherService
           }
         })
       )
+
+      const workspace = this.workspaceService.currentWorkspace
+
+      if (workspace.value?.type === "team") {
+        this.fetchTeamEnvironmentList(workspace.value.teamID, (results) => {
+          this.teamEnvironmentList = results
+          minisearch.addAll(
+            results.map((entry) => {
+              let id = `team-environment-${entry.teamID}:${entry.id}`
+
+              if (
+                this.selectedEnvIndex.value?.type === "TEAM_ENV" &&
```

Comment:
Is the truthy check necessary?

Reply:
Yeah, it's because of the type fail issue. `teamEnvID` only available of the type is `TEAM_ENV`